### ISSUE TITLE
Shen's Last Gift cosmetic options slider is displayed as "DLC_3"

### ISSUE THREAD

**Issue by Iridar** [X2CommunityCore/X2WOTCCommunityHighlander#1111]
On the settings page, the game displays sliders to control the appearance frequency of cosmetic parts from a certain mod or DLC. 

Original code is arguably bugged, and does not display the slider if the DLC name is missing localization.

Highlander's issue #150 (I think?) fixes that problem, and makes the game display the slider using the DLC name itself if the localized DLC name is unavailable. 

This makes the entry for Shen's Last Gift appear as "DLC_3", which apparently scares some people into thinking it's a bug and a problem that needs to be solved ([example](https://github.com/long-war-2/lwotc/issues/1469)).

[Example Image](https://user-images.githubusercontent.com/11407231/140824512-b867fa17-822d-4e64-b845-d721779a0674.jpg)

Personally I don't think it's a problem worth scratching, but if anybody wants to fix it, it can be fixed by Highlander by shipping a localized DLC name for Shen's Last Gift. 

Keeping in mind that even showing a slider for the SLG is of arguable usefulness, since the only cosmetics added by that mod are for SPARKs, so unless the player has cosmetic mods that give SPARKs a whole new appearance, the slider will be useless. And even then, I'm not sure setting the slider to 0 would be useful, since SPARKs generated by the game may or may not use SLG body parts by default either way.

**Comment by TheRealAyCe** [comment]
Either hide the slider if changing it has no effect anyways, or make it at least display a name. Showing a cryptic unlocalized string helps nobody.

**Comment by Iridar** [comment]
> Either hide the slider if changing it has no effect anyways, or make it at least display a name. Showing a cryptic unlocalized string helps nobody.

The slider is displayed for all mods that add cosmetic body parts marked with a DLC name - that's the primary purpose of that part of Highlander's functionality. The Shen's Last Gift is not special handled, it's not like Highlander goes out of its way to display the slider for it. 

Highlander is a community project. It you wish to see an improvement happen, you're welcome to submit a pull request with this change.

**Comment by TheRealAyCe** [comment]
But you agree that a cryptic unlocalized string is unhelpful and should be fixed?

I would love to help, but, as I'm sure you know, it's not possible to just contribute to a project when you have no idea how to even start. If the fix was as simple as just adding a line to a localization file, I'm sure you would have done that yourself already.

**Comment by Iridar** [comment]
> But you agree that a cryptic unlocalized string is unhelpful and should be fixed?

Sure, otherwise I wouldn't have opened the issue. 

> I would love to help, but, as I'm sure you know, it's not possible to just contribute to a project when you have no idea how to even start. 

I suggest you start by downloading XCOM 2 WOTC SDK. 

> If the fix was as simple as just adding a line to a localization file, I'm sure you would have done that yourself already.

It should indeed be that easy, but there are a thousand other things I'd rather do first. I might have even done it myself, but your accusatory tone and general lack of politeness just rub me the wrong way.

**Comment by TheRealAyCe** [comment]
> Highlander is a community project. It you wish to see an improvement happen, you're welcome to submit a pull request with this change.

> I suggest you start by downloading XCOM 2 WOTC SDK.

So do you expect me to download the whole SDK just for changing a single line in a config file, a change which most likely could be made directly here in Github, if you'd know where to look? I won't do that.

> It should indeed be that easy, but there are a thousand other things I'd rather do first.
You mean instead of fixing the localization, or a better way to fix it? If you're talking about better ways, I don't know them. Don't expect others to be able to read your mind. I gather from this that it would be a bad idea for anyone but yourself to attempt to do this, so another reason why I won't. If the fix is indeed this easy, it should be about a minute of your time. Think about that and about people using your mod wondering whether there is a bug on their end. You already know this quirk, us players don't.

> I might have even done it myself, but your accusatory tone and general lack of politeness just rub me the wrong way.
I did not accuse anyone of anything, and if you think I lack politeness then please point out to me what you mean by that. I simply want the error fixed. As I said, I would even do it myself (which I've done for other issues in the past), but for this issue I don't know how. So does that mean you won't do it because you think I didn't express myself nice enough? At least I reported it. You rather people don't?

**Comment by Xymanek** [comment]
Everyone here are volunteers. As such, your options are:

* Implement the fix/improvement yourself and submit it
* Leave the issue open until someone feels like addressing it

Attempting to persuade someone to do something will, at best, get you nowhere

**Comment by Iridar** [comment]
> So do you expect me to download the whole SDK just for changing a single line in a config file, a change which most likely could be made directly here in Github, if you'd know where to look? I won't do that.

Lacking in personal skills or time to fix a presented issue is common. If you don't want to handle the problem yourself, then do what everybody else in that position does - wait for somebody else to do so. 

It's not like you personally or me personally are the only two people that could ever potentially do this. This is why this issue exists - to inform about the problem that may or may not be resolved at some point.

> You mean instead of fixing the localization, or a better way to fix it? 

Instead of. Like many other modmakers, I have personal projects I want to do, not to mention a life to live. 

> I gather from this that it would be a bad idea for anyone but yourself to attempt to do this, so another reason why I won't. If the fix is indeed this easy, it should be about a minute of your time. Think about that and about people using your mod wondering whether there is a bug on their end. You already know this quirk, us players don't.

I think you misunderstand how it all works. Like I said, Highlander is a community project. It's not "my" mod, I'm just one of the dozens on people that work or worked on it. Many of those people are capable of fixing this problem should they decide to donate their time and effort for it.

You are the only person so far to mention this issue that I've seen, and at absolute worst the effect on the player is _"Hmm I don't know what that is"_. Which is why I said in the opening post that I don't view this problem as an important one.

> I did not accuse anyone of anything, and if you think I lack politeness then please point out to me what you mean by that. 

"Either hide the slider if changing it has no effect anyways, or make it at least display a name. Showing a cryptic unlocalized string helps nobody."

I'm talking about this specific statement. Here you acted like this problem is the direct result of Highlander doing something bad. When in actuality it is - at worst - a confusing side effect caused by Firaxis not shipping a localized DLC name for Shen's Last Gift, and induced by an actually useful piece of Highlander's functionality that is working as intended. 

Of course, it's hard to read tone over the internet, but you certainly didn't go out of your way to present your suggestions / concerns in a polite manner.

As one of Highlander developers, I've already done what could be reasonably expected from me - saw your report, explained the problem to you, and created a tracking issue so that it maybe can be resolved in the future, but then you went on the offensive, instead of just asking nicely.

**Comment by TheRealAyCe** [comment]
Whatever. I won't report any more errors, as you guys clearly seem more interested in starting a flame war than actually fixing problems.

**Comment by RowanSkie** [comment]
Has anyone tried to solve this problem? Since it only appears when you get this mod active and notice someone complaining about it so...

does anyone know the localization string that checks for this so I can try to create a small patch?

**Comment by Iridar** [comment]
I think this is handled by `PartContentLabel` in `X2DownloadableContentInfo`.

So adding a `DLC_3.int` file with the following:
```
[X2DownloadableContentInfo_DLC_Day90]
PartContentLabel = "Shen's Last Gift" 
```
should do it. Not 100% sure here, so testing to confirm is required.

Bonus points for adding localized labels for all languages. You can copy them from the actual DLC localization itself, it probably exists somewhere in DLC localization folders.

**Comment by RowanSkie** [comment]
> I think this is handled by `PartContentLabel` in `X2DownloadableContentInfo`.
> 
> So adding a `DLC_3.int` file with the following:
> 
> ```
> [X2DownloadableContentInfo_DLC_Day90]
> PartContentLabel = "Shen's Last Gift" 
> ```
> 
> should do it. Not 100% sure here, so testing to confirm is required.
> 
> Bonus points for adding localized labels for all languages. You can copy them from the actual DLC localization itself, it probably exists somewhere in DLC localization folders.

It actually worked. I'm gonna start doing the other languages, and probably the mouse-over text that appears.

**Comment by Iridar** [comment]
> It actually worked. I'm gonna start doing the other languages, and probably the mouse-over text that appears.

Ah good. Mouseover text is probably handled by `PartContentSummary`, which you can set right next to `PartContentLabel`

**Comment by RowanSkie** [comment]
Well, so far here's what DLC_3.int has on mine:

```
[X2DownloadableContentInfo_DLC_Day90]
PartContentLabel = "Shen's Last Gift"
PartContentSummary="Adjust the likelyhood that new SPARK units will appear with customization options"
```

Not exactly fancy mouseover, since I have no idea what the slider does. 

(Also base game DLCs have likelyhood instead of likelihood. Woah.)

**Comment by Iridar** [comment]
Yeah that's what slider does, though as I mention in the opening post, it remains to be checked if the slider even does anything, even in the presence of mods that add new cosmetic body parts for SPARKs.

**Comment by Iridar** [comment]
Since this issue pertains strictly to a DLC, it should not be addressed in the main Highlander. If ever make a Shen's Last Gift Highlander, we'll fix it there. For now I'll put the fix into Core Collection Meta Mod.
